packages/monaco/src/browser/monaco-workspace.ts

```diff
@@ -259,6 +270,13 @@ export class MonacoWorkspace {
         }
     }
 
+    protected saveAll(resources: Set<string>): void {
+        for (const uri of resources.values()) {
+            this.textModelService.get(uri)?.save();
+        }
+
+    }
```

Comment:
Suggestion: We should await all of these (in parallel)

```suggestion
    protected async saveAll(resources: Set<string>): Promise<void> {
        await Promise.all(Array.from(resources.values()).map(uri => this.textModelService.get(uri)?.save()));
    }
```